src/main/webapp/placeGuides/MapPlaceGuideDisplayer.js

```diff
@@ -6,12 +6,81 @@ class MapPlaceGuideDisplayer {
     this._placeGuidesOnMap = {};
   }
 
+  static lngDistance(lngWest, lngEast) {
+    if (lngWest < lngEast) {
+      return lngEast - lngWest;
+    } else {
+      // West points distance from IDL + east points distance to IDL;
+      // (180 - lngWest) + (180 + lngEast);
+      return 360 - lngWest + lngEast;
+    }
+  }
+
+  // This function scans all the placeGuides to find the minimum map area
+  // which contains all of them.
+  // To do so, it finds the minimum and maximum latitude coordinate of all
+  // placeGudies. These will be set as the southern and northern bound of the
+  // map area.
+  // For longitudes, it firs finds the maximum lane which doesn't contain any
+  // placeGuides(for this purpose, placeGuides are sorted based on their
+
+  // longitude coordinates.
+  static comparePlaceGuidesOnMap(placeGuideA, placeGuideB) {
+    const positionA = placeGuideA.marker.getPosition();
+    const positionB = placeGuideB.marker.getPosition();
+    return positionA.lng() - positionB.lng();
+  }
+
   update(placeGuides) {
     this.removePreviousPlaceGuidesFromMap(placeGuides);
     this.addNewPlaceGuidesToMap(placeGuides);
     this.updateMarkerClusters();
   }
 
+  // Sorts placeGuides in an ascending order based on their
+
+  // longitude coordinate) and then complements it.
+  adjustMapToShowAll() {
+    let minLat = 90;
+    let maxLat = -90;
+    let maxLngDifference = 0;
+    let maxLngDifferenceWestCorner = 0;
+    let nextMarkerLng;
+    const placeGuidesOnMap = Object.values(this._placeGuidesOnMap);
+    placeGuidesOnMap.sort(MapPlaceGuideDisplayer.comparePlaceGuidesOnMap);
+    if (placeGuidesOnMap.length > 0) {
+      for (let i = 0; i < placeGuidesOnMap.length; i++) {
+        const position = placeGuidesOnMap[i].marker.getPosition();
+        minLat = Math.min(position.lat(), minLat);
+        maxLat = Math.max(position.lat(), maxLat);
+        if (i + 1 < placeGuidesOnMap.length) {
+          nextMarkerLng =
+              placeGuidesOnMap[i + 1].marker.getPosition().lng();
+        } else {
+          nextMarkerLng = placeGuidesOnMap[0].marker.getPosition().lng();
+        }
+        if (MapPlaceGuideDisplayer.lngDistance(position.lng(),
+            nextMarkerLng) >
+            maxLngDifference) {
+          maxLngDifference =
+              MapPlaceGuideDisplayer.lngDistance(position.lng(),
+                  nextMarkerLng);
+          maxLngDifferenceWestCorner = position.lng();
+        }
+      }
+      let westLng = maxLngDifferenceWestCorner + maxLngDifference;
+      if (westLng > 180) westLng -= 360;
+      const eastLng = maxLngDifferenceWestCorner;
+      const southWestCorner = new google.maps.LatLng(minLat, westLng);
+      const northEastCorner = new google.maps.LatLng(maxLat, eastLng);
+      map.setZoom(15);
+      map.fitBounds(
+          new google.maps.LatLngBounds(southWestCorner,
+              northEastCorner),
+          10);
+    }
```

Comment:
This function is a little long. Can we try to break into multiple smaller ones for readability?

Reply:
Done.

Reply:
+1. I think we can try to simplify the logic as well. What about normalizing all the degree values so they are in the range [-180, 180] or some other canonical range. Then we get min and max as we do with lat. Finally we just need to check if (max - min) > 180 and if that is the case switch east and west. Maybe some of that is even already handled by the fitBounds method.

Reply:
Update: we discussed that the solution mentioned above does not cover all the cases(for example, consider the PlaceGuides at longitudes -170, -160, 160, 170: the algorithm above would give the solution [170, -170], but this one doesn't contain all PlaceGuides. The correct solution would be [160, -160].
Instead, a new, simpler and less error-prone approach was implemented which relies on the extend() method of LatLngBounds, provided by the Maps API.

Reply:
Remark: this solution(the one based on the extend() function) won't always lead to an optimal solution, in contrast with the original algorithm based on sorting.